pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -199,128 +191,37 @@ private InstanceResponseBlock executeInternal(ServerQueryRequest queryRequest, E
       return instanceResponse;
     }
 
-    TableDataManager tableDataManager = _instanceDataManager.getTableDataManager(tableNameWithType);
-    if (tableDataManager == null) {
-      String errorMessage = "Failed to find table: " + tableNameWithType + " on server: "
-          + _instanceDataManager.getInstanceId();
+    TableExecutionInfo executionInfo;
+    try {
+      executionInfo =
+          SingleTableExecutionInfo.create(_instanceDataManager, tableNameWithType, queryRequest.getSegmentsToQuery(),
+              queryRequest.getOptionalSegments(), queryContext);
+    } catch (NoSuchElementException ne) {
```

Comment:
nit: Rather than throwing and catching an unchecked / runtime exception, I think it'd be nicer if we could return `null` from `SingleTableExecutionInfo.create` (and annotate it as nullable) in such scenarios; alternatively, we could throw a checked exception.

Reply:
I've changed it to throw `TableNotFoundException`